server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchema.java

```diff
@@ -0,0 +1,273 @@
+ * <ul>
+ *   <li>60-second refresh interval for schema change detection</li>
+ *   <li>49-second monitor interval (chosen to avoid harmonics with the 60s refresh cycle)</li>
+ * </ul>
+ * <p>
+ * This component is essential for CDC operations as it ensures that CDC consumers always have
+ * up-to-date schema information, enabling proper data serialization, deserialization, and
+ * processing across schema evolution events.
+ * <p>
+ * This class is thread-safe and designed as a singleton for dependency injection into CDC
+ * and other schema-dependent components.
+ *
+ * @see org.apache.cassandra.sidecar.db.CdcDatabaseAccessor
+ * @see org.apache.cassandra.bridge.CdcBridge
+ * @see org.apache.cassandra.spark.data.CqlTable
+ */
+@Singleton
+public class CassandraClusterSchema implements PeriodicTask
+{
+    // 49sec least-common multiple with 60sec is 49min so offers best monitor frequency without clashing with 60sec
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraClusterSchema.class);
+
+    private final AtomicReference<String> currSchemaText = new AtomicReference<>("");
+    private final AtomicReference<Set<CqlTable>> cdcTables = new AtomicReference<>(Collections.emptySet());
+    private final ConcurrentHashMap<TableIdentifier, UUID> tableIdCache = new ConcurrentHashMap<>();
+    private final CdcDatabaseAccessor databaseAccessor;
+    private final CopyOnWriteArrayList<Runnable> schemaChangeListeners = new CopyOnWriteArrayList<>();
+    private final SidecarConfiguration sidecarConfiguration;
+    private final InstanceMetadataFetcher instanceFetcher;
+    private final SecondBoundConfiguration tableSchemaRefreshTime;
+    private final CassandraBridgeFactory cassandraBridgeFactory;
+
+    public CassandraClusterSchema(InstanceMetadataFetcher instanceFetcher,
+                                  CdcDatabaseAccessor databaseAccessor,
+                                  SidecarConfiguration sidecarConfiguration,
+                                  CassandraBridgeFactory cassandraBridgeFactory)
+    {
+
+        this.instanceFetcher = instanceFetcher;
+        this.databaseAccessor = databaseAccessor;
+        this.sidecarConfiguration = sidecarConfiguration;
+        this.tableSchemaRefreshTime = sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime();
+        this.cassandraBridgeFactory = cassandraBridgeFactory;
+    }
+
+    public void addSchemaChangeListener(Runnable listener)
+    {
+        schemaChangeListeners.add(listener);
+    }
+
+    public void refresh()
+    {
+        NodeSettings nodeSettings = instanceFetcher.callOnFirstAvailableInstance(instance-> instance.delegate().nodeSettings());
+        CassandraBridge cassandraBridge = cassandraBridgeFactory.get(nodeSettings.releaseVersion());
+        CdcBridge cdcBridge = CdcBridgeFactory.getCdcBridge(cassandraBridge);
+
+        try
+        {
+            LOGGER.info("Checking for schema changes...");
```

Comment:
this at debug level?
```suggestion
            LOGGER.debug("Checking for schema changes...");
```